### ISSUE TITLE
PMAT results in 0

### ISSUE THREAD

**Issue by Yoavp81** [DriesLanduyt/PMAT#9]
Hello,

Until recently, I was able to run PMAT. However, now, when I run it, all values are set at 0.
The only change I've made that I can think of is updating QGIS to the latest version.
I've tried uninstalling QGIS, revert back to the previous version and install PMAT once again, but the results are still 0, even for the example file.

Could you please help?

Thank you,
Yoav

**Comment by DriesLanduyt** [comment]
Hi Yoav, 

As I haven't tested my plugin on different QGIS versions, it's very hard to tell what caused this issue, especially since no errors pop up. Does the "slow" mode returns you an error?

**Comment by Yoavp81** [comment]
Yes, it returned to following error on "slow" mode:
![image](https://user-images.githubusercontent.com/36417074/36149966-86a50d02-10ca-11e8-8704-8bfe913f5f35.png)

**Comment by DriesLanduyt** [comment]
Difficult one to deal with. Strange that reverting your update didn't help. Don't have any clue at the moment...

**Comment by Yoavp81** [comment]
Thanks anyway!

**Comment by leilamac** [comment]
Ok unfortunately I am having this problem too. I am a beginner coder so still figuring out how to debug a QGIS plugin, but given the location that this error crops up and the wording you have included, it suggests that the model is either not open, or not running/returning output beliefs. 

So, just to be sure that I have set everything up correctly with the plugin inputs - what exactly should the text be in the "{tick box} Insert Netica AP" dialogue box? I am unable to find a description of what exactly is expected and what the impact of including this/vs. not including this is. And where do you keep the Netica.exe vs. the Netica.dll vs. the plugin PMAT files vs. the input files/BN? I have followed your instructions for the dll in the bin, eg, but must I also copy the Netica.lib somewhere? Thank you for clarifying and apologies for bugging you.

**Comment by DriesLanduyt** [comment]
Just found part of the problem, the IOError pops up because the code tries to print something to the console which isn't possible while running in QGIS. Have now deleted all "print" commands in the code so this should be solved. 

This, however, does not solve the real problem, being a zero belief vector returned by the model. Have looked into this a while ago but unfortunately couldn't find the reason for this behavior. 

To respond to your question on file locations, you only need to make sure that both the spatial input data and the network file are stored in the same folder. Concerning the thick box, you can ignore that one. I have never tested this thoroughly.

**Comment by ChristaGitHub** [comment]
Hello,

I am an intermediate GIS-user and rather new to BBN, so your plugin would be perfect for me.
But I have the same problem as Yoav, already with your example data. It only returns 0 values for all outputs. 
![zero_values](https://user-images.githubusercontent.com/45963028/50141535-08635680-02a8-11e9-835f-80cb144b3076.JPG)

**Comment by ChristaGitHub** [comment]
So I am rather lost, have you any idea?
I have installed qgis version 2.18, Netica version 605

Thanks in advance,
Christa

**Comment by DriesLanduyt** [comment]
No, I'm sorry, haven't had the time to look into that. If you're into Python programming you could just use the raw code to do similar things but I'm afraid it's the .dll file (and a newer Python version than the one I used during programming) what is causing this problem. If that's the case you will have to go via the Netica interface (or other BBN software), a little more steps to complete but definitely feasible as well. Feel free to send me an email if you want some advice on that. 

Best regards, 
Dries
